jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileSsoSessionProvider.kt

```diff
@@ -0,0 +1,75 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.credentials.profiles
+
+import software.amazon.awssdk.auth.credentials.AnonymousCredentialsProvider
+import software.amazon.awssdk.auth.credentials.AwsCredentials
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider
+import software.amazon.awssdk.profiles.Profile
+import software.amazon.awssdk.profiles.ProfileFile
+import software.amazon.awssdk.profiles.ProfileProperty
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.sso.SsoClient
+import software.amazon.awssdk.services.ssooidc.SsoOidcClient
+import software.amazon.awssdk.utils.SdkAutoCloseable
+import software.aws.toolkits.jetbrains.core.AwsClientManager
+import software.aws.toolkits.jetbrains.core.credentials.diskCache
+import software.aws.toolkits.jetbrains.core.credentials.sso.SsoAccessTokenProvider
+import software.aws.toolkits.jetbrains.core.credentials.sso.SsoCredentialProvider
+import java.util.Optional
+
+class ProfileSsoSessionProvider(profile: Profile) : AwsCredentialsProvider, SdkAutoCloseable {
+    private val ssoClient: SsoClient
+    private val ssoOidcClient: SsoOidcClient
+    private val credentialsProvider: SsoCredentialProvider
+    private val mySsoSection: SsoMetadata
+    init {
+        val ssoSession = profile.requiredProperty(SsoSessionConstants.PROFILE_SSO_SESSION_PROPERTY)
+        val ssoSessionSection: Optional<Profile>? = ProfileFile.defaultProfileFile().getSection(SsoSessionConstants.SSO_SESSION_SECTION_NAME, ssoSession)
```

Comment:
should be reading out of the profileholder snapshot instead of directly from the file

Reply:
Profile holder is currently holding only the list of profiles. Have verified that sso-session is not stored as a profile inside profile holder. So I had to read from ProfileFile.

Reply:
will tackle this in Seperate PR